File: SimpleEQClient/SimpleEQClient/Assets/Scripts/Objects/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;
using Objects.Data;

namespace Objects
{
    public class Player : Object
    {
        public uint level;
        public ulong exp;

        public short max_hp;
        public short curr_hp;
        public uint base_damage;

        public Dictionary<uint, Faction> factions;
        public List<Quest> quests;
    }
}

File: WorldServer/Player.cpp
#include "Player.h"

Player::~Player()
{
	view_list.clear();
}

void Player::GainExp(unsigned long _exp)
{
	exp += _exp;
}

void Player::IncreaseHP(unsigned short hp)
{
	if (curr_hp + hp > max_hp)
		curr_hp = max_hp;
	else
		curr_hp += hp;
}

void Player::Update()
{
}
